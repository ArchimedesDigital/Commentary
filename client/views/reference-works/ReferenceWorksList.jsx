import Masonry from 'react-masonry-component/lib';


ReferenceWorksList = React.createClass({

	propTypes: {
		commenterId: React.PropTypes.string,
	},

	// This mixin makes the getMeteorData method referenceWork
	mixins: [ReactMeteorData],

	// Loads items from the referenceWorks collection and puts them on this.data.referenceWorks
	getMeteorData() {
		// SUBSCRIPTIONS:
		const query = {};
		if (this.props.commenterId) {
			query.authors = this.props.commenterId;
			Meteor.subscribe('referenceWorks.commenterId', this.props.commenterId);
		} else {
			Meteor.subscribe('referenceWorks');
		}

		// FETCH DATA:
		const referenceWorks = ReferenceWorks.find(query, { sort: { title: 1 } }).fetch();

		return {
			referenceWorks,
		};
	},

	renderReferenceWorks() {
		return this.data.referenceWorks.map((referenceWork, i) => (
			<ReferenceWorkTeaser
				key={i}
				referenceWork={referenceWork}
			/>
		));
	},

	render() {
		const masonryOptions = {
			isFitWidth: true,
			transitionDuration: 300,
		};
		return (
			<div>
				{this.data.referenceWorks.length ?
					<Masonry
						options={masonryOptions}
						className="reference-works-list"
					>
						{this.renderReferenceWorks()}
					</Masonry>
					:
					<p className="no-results no-results-reference-works">No reference works found.</p>
				}
			</div>
		);
	},

});
