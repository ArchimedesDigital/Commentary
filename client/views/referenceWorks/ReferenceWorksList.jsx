ReferenceWorksList = React.createClass({

	propTypes: {
		type: React.PropTypes.string.isRequired,
	},

	// This mixin makes the getMeteorData method referenceWork
	mixins: [ReactMeteorData],

	// Loads items from the referenceWorks collection and puts them on this.data.referenceWorks
	getMeteorData() {
		const query = {
			type: this.props.type,
			count: { $gt: 0 },
		};

		return {
			referenceWorks: ReferenceWorks.find(query, { sort: { title: 1 } }).fetch(),
		};
	},

	renderReferenceWorks() {
		return this.data.referenceWorks.map((referenceWork, i) => {
			return (
				<ReferenceWorkTeaser
					key={i}
					referenceWork={referenceWork}
				/>
			);
		});
	},

	render() {
		return (
			<div className="reference-works-list">
				{this.renderReferenceWorks()}
			</div>
		);
	},

});
