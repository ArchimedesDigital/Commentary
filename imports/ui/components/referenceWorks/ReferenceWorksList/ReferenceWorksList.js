import React from 'react';
import PropTypes from 'prop-types';
import { Meteor } from 'meteor/meteor';
import { Session } from 'meteor/session';
import { createContainer } from 'meteor/react-meteor-data';

import Masonry from 'react-masonry-component/lib';

// models
import ReferenceWorks from '/imports/models/referenceWorks';

// components
import ReferenceWorkTeaser from '/imports/ui/components/referenceWorks/ReferenceWorkTeaser';

// lib
import Utils from '/imports/lib/utils';


class ReferenceWorksList extends React.Component {

	renderReferenceWorks() {
		return this.props.referenceWorks.map((referenceWork, i) => (
			<ReferenceWorkTeaser
				key={i}
				referenceWork={referenceWork}
			/>
		));
	}

	render() {
		const { referenceWorks } = this.props;
		const masonryOptions = {
			isFitWidth: true,
			transitionDuration: 300,
		};

		if (!referenceWorks) {
			return null;
		}

		return (
			<div>
				{referenceWorks.length ?
					<Masonry
						options={masonryOptions}
						className="reference-works-list"
					>
						{this.renderReferenceWorks()}
					</Masonry>
					:
					<p className="no-results no-results-reference-works">No reference works found.</p>
				}
			</div>
		);
	}

}

ReferenceWorksList.propTypes = {
	commenterId: PropTypes.string,
	referenceWorks: PropTypes.array,
};

const ReferenceWorksListContainer = createContainer(({ commenterId }) => {
	// SUBSCRIPTIONS:
	const query = {};
	if (commenterId) {
		query.authors = commenterId;
		Meteor.subscribe('referenceWorks.commenterId', commenterId, Session.get('tenantId'));
	} else {
		Meteor.subscribe('referenceWorks', Session.get('tenantId'));
	}

	// FETCH DATA:
	const referenceWorks = ReferenceWorks.find(query, { sort: { title: 1 } }).fetch();

	return {
		referenceWorks,
	};
}, ReferenceWorksList);

export default ReferenceWorksListContainer;
