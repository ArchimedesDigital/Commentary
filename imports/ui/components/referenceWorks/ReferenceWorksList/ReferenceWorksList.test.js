import React from 'react';
import renderer from 'react-test-renderer';

// component:
import ReferenceWorksList from './ReferenceWorksList';

describe('ReferenceWorksList', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(
				<ReferenceWorksList
					referenceWorks={[]}
				/>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
