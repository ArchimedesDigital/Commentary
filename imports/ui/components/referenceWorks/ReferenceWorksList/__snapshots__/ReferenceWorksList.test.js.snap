// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReferenceWorksList renders correctly 1`] = `
<div>
  <p
    className="no-results no-results-reference-works"
  >
    No reference works found.
  </p>
</div>
`;
